game/hud/src/widgets/Crafting/components/NameInput/index.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * @Author: Mehuge (mehuge@sorcerer.co.uk)
+ * @Date: 2017-05-13 18:19:58
+ * @Last Modified by: Mehuge (mehuge@sorcerer.co.uk)
+ * @Last Modified time: 2017-05-24 19:55:40
+ */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import Select from '../Select';
+import Label from '../Label';
+import Input from '../Input';
+import { GlobalState } from '../../services/session/reducer';
+import { StyleSheet, css, merge, nameInput, NameInputStyles } from '../../styles';
+
+export interface NameInputReduxProps {
+  dispatch?: (action: any) => void;
+  name?: string;
+  style?: Partial<NameInputStyles>;
+}
+
+export interface NameInputProps extends NameInputReduxProps {
+  onChange: (name: String) => void;
+}
+
+interface NameInputState {}
+
+const select = (state: GlobalState, props: NameInputProps) : NameInputReduxProps => {
+  return {
+    name: state.job.name,
+  };
+};
+
+const NameInput = (props: NameInputProps) => {
+  const ss = StyleSheet.create(merge({}, nameInput, props.style));
+  return (
+    <div className={'name-input ' + css(ss.container)}>
```

Comment:
Ah ha... well.  I have a very good reason for doing it, perhaps you will agree.  One thing I absolutely HATE about aphrodite and the way we use it is we end up with a million `<div class="container_f8sfs9fs8">` elements and it is a right pain in the backside to debug, because it's not obvious what 'component' it is linked to.

So I first searched the aphrodite docs to see if there was a solution and came across an issue reported essentially saying the same, but aphrodite doesn't or hasn't provided a solution, they don't consider it a problem.

I then thought well I can just give the elements IDs but didn't really like that, so then it occurred to me there is absolutely no reason I can't give the components human readable class names as well as the aphrodite ones.

So that's what I did and it's brilliant!  So much easier to debug.